ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFToUnixTimeStamp.java

```diff
@@ -117,6 +121,19 @@ protected void initializeInput(ObjectInspector[] arguments) throws UDFArgumentEx
             + " takes only string/date/timestamp/timestampwltz types. Got Type:" + arg1OI
             .getPrimitiveCategory().name());
     }
+
+    if (timeZone == null) {
+      timeZone = SessionState.get().getConf().getLocalTimeZone();
+      formatter.setTimeZone(TimeZone.getTimeZone(timeZone));
+    }
+  }
+
+  public void configure(MapredContext context) {
```

Comment:
No, this is the comment of the super method that it overrides (I added the override notation since it was missing):
```
  /**
   * Additionally setup GenericUDF with MapredContext before initializing.
   * This is only called in runtime of MapRedTask.
   *
   * @param context context
   */
```
Basically this is needed because we need to set up the timezone from the configuration when we are about to execute the udf in each of the nodes, but we only have access to the configuration object through this context. This is called before calling the initialize method.